test/shopify_app/controller_concerns/login_protection_test.rb

```diff
@@ -255,4 +288,12 @@ def with_application_test_routes
       yield
     end
   end
+
+  def with_custom_login_url(url)
+    original_url = ShopifyApp.configuration.login_url.dup
+
+    ShopifyApp.configure { |config| config.login_url = url }
+    yield
+    ShopifyApp.configure { |config| config.login_url = original_url }
```

Comment:
I believe this last line should be behind `ensure` so the url is restored even if an exception occurs

Reply:
Ah yep, thanks!